File: ai-hub-backend/src/routes/adminRoutes.js
import express from 'express';
import adminContainer from '../di/admin/rootAdminContainer.js';

const router = express.Router();
const controller = adminContainer.adminController;

router.post('/register', controller.register.bind(controller));
router.post('/verify', controller.verifyEmail.bind(controller));
router.post('/login', controller.login.bind(controller));
router.post('/forgot-password', controller.requestPasswordReset.bind(controller));
router.post('/verify-reset-code', controller.verifyResetCode.bind(controller));
router.post('/set-new-password', controller.setNewPassword.bind(controller));

export default router;

File: backend/infra/http/gin/routes/admin_routes.go
package routes

import (
	"ai_hub.com/app/infra/http/gin/controllers"
	"github.com/gin-gonic/gin"
)

func AdminRoutes(r gin.IRouter, controller *controllers.AdminController) {
	admin := r.Group("/admin")
	{
		admin.POST("/login", controller.Login)
		admin.POST("/register", controller.Register)
		admin.POST("/:id/verify", controller.Verify)
		admin.POST("/reset/start", controller.StartReset)
		admin.POST("/reset/confirm", controller.ConfirmReset)
		admin.POST("/reset/change", controller.ChangePassword)
		admin.PATCH("/:id/rename", controller.Rename)
		admin.DELETE("/:id", controller.Delete)
	}
}

type AdminRouter struct {
	controller *controllers.AdminController
}

func NewAdminRoutes(controller *controllers.AdminController) *AdminRouter {
	return &AdminRouter{controller: controller}
}

func (ar *AdminRouter) Mount(r gin.IRouter) {
	AdminRoutes(r, ar.controller)
}
